### PR TITLE
feat: added redo action to adv payments

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- The member field should also be copied over when redoing an simple payment action.
- Redo action should be visible in the activity feed kebab menu as well.
- Redo action should be available for all Advanced payment options, including Staged and Split.

## Testing

* Check if the action is 'copied' after selecting redo action option from meatball menu in activity table

## Diffs

**New stuff** ✨

* Added redo action option to activity table on homepage and activity page

Resolves https://github.com/JoinColony/colonyCDapp/issues/2840

### CODE REVIEW

**Review by arrenv — CHANGES_REQUESTED** [review]
@CzarekDryl The redo is working great, nice one!!

I noticed a couple of things, they may or may not be elated to this functionality, but raising here just in case it is.

1. If you have Split payment that is being re-done that has odd fractions, you get a long value that appears. I can understand that this is probably how this feature is able to work, but, it could be worth going the other way, so that you are spending less, rather then more then the entered value. NOTE: Removing this requirement to it's own issue here - https://github.com/JoinColony/colonyCDapp/issues/3012

[split-redo.webm](https://github.com/user-attachments/assets/8572cf70-5aad-49ea-b201-53aa2858c139)

2. If you redo an action then change the action type to another payment action, including simple payment, changing the amount field auto triggers the not enough token validation incorrectly.

[validation-redo.webm](https://github.com/user-attachments/assets/68ed14d8-ff56-4d20-b8ec-8317f85f7f94)

**Comment by CzarekDryl** [comment]
@arrenv Everything should work fine now, please check it :)

**Review by arrenv — APPROVED** [review]
@CzarekDryl Looks to be working great, thank you!

**Comment by arrenv** [comment]
Just want to note that not this PR, but this branch has re-introdced the width issue on the amount field if you type anything but a number in.

image

**Review by jakubcolony — CHANGES_REQUESTED** [review]
@CzarekDryl please try to create a payment with the following values:
image

The app crashes whenever I try to redo such payment:

https://github.com/user-attachments/assets/7a07b126-27f0-43e1-a5c6-3de15af7b801

**Comment by CzarekDryl** [comment]
@jakubcolony This should work as expected now.

https://github.com/user-attachments/assets/60cdaba3-1642-4857-963b-60061974c556

File: src/components/v5/frame/ColonyHome/ColonyHome.tsx [hunk 1]
```diff
@@ -32,6 +33,10 @@ const ColonyHome = () => {
   const isMobile = useMobile();
   const selectedDomain = useGetSelectedDomainFilter();
   const teamsBreadcrumbs = useCreateTeamBreadcrumbs();
+  const [selectedAction, setSelectedAction] = useState<string | undefined>(
+    undefined,
+  );
+  const { defaultValues } = useGetActionData(selectedAction || undefined);
```
Comment (jakubcolony) [inline src/components/v5/frame/ColonyHome/ColonyHome.tsx:39]: Do we need two such states, in both `ColonyHome` and `ActivityPage` or could it live in `ColonyActionsTable` instead?

**Review by jakubcolony — APPROVED** [review]
Tested with different values including min/max allowed, permissions/staking decision method and split & staged payments, all of which seem to be working fine:

image

image

image

Nice work @CzarekDryl

File: src/components/v5/frame/ColonyHome/ColonyHome.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/v5/frame/ColonyHome/ColonyHome.tsx:39]: @jakubcolony I tried to add it to the`ColonyActionsTable` but there was a problem with rerenders

**Pull request merged**